app/models/miq_dialog.rb

```diff
@@ -1,30 +1,35 @@
 class MiqDialog < ApplicationRecord
-  validates_presence_of   :name, :description
-  validates_uniqueness_of :name, :scope => :dialog_type, :case_sensitive => false
-
-  DIALOG_DIR = Rails.root.join("product/dialogs/miq_dialogs")
+  validates :name, :description, :presence => true
+  validates :name, :uniqueness => { :scope => :dialog_type, :case_sensitive => false }
 
   DIALOG_TYPES = [
     [_("VM Provision"),                "MiqProvisionWorkflow"],
     [_("Configured System Provision"), "MiqProvisionConfiguredSystemWorkflow"],
     [_("Host Provision"),              "MiqHostProvisionWorkflow"],
     [_("VM Migrate"),                  "VmMigrateWorkflow"],
-  ]
+  ].freeze
 
   serialize :content
 
   def self.seed
-    sync_from_dir
+    sync_from_dir(Rails.root.join('product', 'dialogs', 'miq_dialogs'))
+    sync_from_plugins
+  end
+
+  def self.sync_from_dir(root)
+    Dir.glob(root.join("*.yaml")).each { |f| sync_from_file(f, root) }
   end
 
-  def self.sync_from_dir
-    Dir.glob(File.join(DIALOG_DIR, "*.yaml")).each { |f| sync_from_file(f) }
+  def self.sync_from_plugins
+    Vmdb::Plugins.instance.registered_provider_plugins.each do |plugin|
+      sync_from_dir(plugin.root.join('content', 'dialogs'))
```

Comment:
do we really want to seed from `content/dialogs` ?
Or is it `content/dialogs/miq_dialogs` ?

TBH I have no idea what both are :/